CONTRIBUTING.md

```diff
@@ -0,0 +1,80 @@
+# Contributing to azure-devops-dependency-visualizer 
+
+Contributions to azure-devops-dependency-visualizer are welcome from all!
+
+azure-devops-dependency-visualizer is managed via [git](https://git-scm.com), with the canonical upstream
+repository hosted on [GitHub](https://github.com/ni/azure-devops-dependency-visualizer/).
+
+azure-devops-dependency-visualizer follows a pull-request model for development.  If you wish to
+contribute, you will need to create a GitHub account, fork this project, push a
+branch with your changes to your project, and then submit a pull request.
+
+See [GitHub's official documentation](https://help.github.com/articles/using-pull-requests/) for more details.
+
+# Getting Started
+
+Install Node.js and run 'npm install' from the src directory in order to install dependencies.  See [Microsoft's documentation](https://docs.microsoft.com/en-us/azure/devops/extend/?toc=%2Fazure%2Fdevops%2Fextend%2Ftoc.json&bc=%2Fazure%2Fdevops%2Fextend%2Fbreadcrumb%2Ftoc.json&view=azure-devops) for information on developing and publishing extensions.
+
+# Before Posting a Pull-Request
+
```

Comment:
I'm not sure the quality level you're shooting for on this repo, but adding a linter is pretty important for JS code since there's no compile step so a syntax error can break your whole site. ESLint is the standard toolchain at NI. The JS coding practices working group is trending towards suggesting the airbnb ruleset as NI's standard, but even the default configuration is better than nothing.

Reply:
> We need to decide if this is going to stay in vanilla js or not. Most Azure DevOps extensions, documentation and tutorials use Typescript so if we expect to attract other Azure DevOps users to contribute it may be worth translating it over - which really isn't that hard. Otherwise some of @jattasNI come into play with the way it is now. The JS, and maybe the CSS, should really be pulled out of the HTML so it can be run through linters and minified for distribution and get a pipeline setup for publishing test & prod stages.

Some of these decisions may just need to be GitHub Issues instead of comments on this PR.